sourced/ml/algorithms/id_splitter/nn_model.py

```diff
@@ -67,59 +93,76 @@ def add_rnn(X: tf.Tensor, units=128, rnn_layer: str=None, dev0: str="/gpu:0",
     :param rnn_layer: type of cell in the RNN.
     :param dev0: device that will be used for forward pass of RNN and concatenation.
     :param dev1: device that will be used for backward pass.
-
     :return: output bidirectional RNN layer.
     """
-    # select the RNN layer
+    # select the RNN layer.
     if rnn_layer is None:
         rnn_layer = DEFAULT_RNN_TYPE
     rnn_layer = getattr(keras.layers, rnn_layer)
 
-    # add the forward & backward RNN
+    # add the forward & backward RNN.
     with tf.device(dev0):
         forward = rnn_layer(units=units, return_sequences=True)(X)
     with tf.device(dev1):
         backward = rnn_layer(units=units, return_sequences=True, go_backwards=True)(X)
 
-    # concatenate
+    # concatenate.
     with tf.device(dev1):
         bidi = Concatenate(axis=-1)([forward, backward])
     return bidi
 
 
-def build_rnn(n_uniq: int, maxlen: int, units: int, stack: int, optimizer: str, dev0: str,
+def build_rnn(maxlen: int, units: int, stack: int, optimizer: str, dev0: str,
               dev1: str, rnn_layer: str=None) -> keras.engine.training.Model:
     """
     Builds a RNN model with the parameters specified as arguments.
 
-    :param n_uniq: number of unique items/characters.
     :param maxlen: maximum length of the input sequence.
     :param units: number of neurons or dimensionality of the output RNN.
     :param stack: number of RNN layers to stack.
     :param optimizer: algorithm to use as an optimizer for the RNN.
-    :param rnn_layer: recurrent layer type to use..
+    :param rnn_layer: recurrent layer type to use.
     :param dev0: first device to use when running specific operations.
     :param dev1: second device to use when running specific operations.
     :return: compiled RNN model.
     """
-    # prepare the model
+    # prepare the model.
```

Comment:
we don't usually put dots in such comments. Otherwise, make it a fully-featured sentence:

```python
# Prepare the models.
```

but it looks even more official.